### PR TITLE
Update resource path to be prefixed with resource name

### PR DESCRIPTION
**Pull request by shashwathi**
Current implementation fetches resources into path `/workspace`. If task has multiple input resources defined without target path, then resource files can overlap.

With this PR input resources will be places under file path `/workspace/task_resource_name` . This will provide specific filepath for each resource and there should be no clash as input resources with same names are not allowed.

**Note** : I have disabled HelmDeployRun e2e test as that requires follow up PR. Test relies on master branch having changes already. It really makes me think there should separate repo with example chart to deploy because we cannot commit change and updated test in same PR. 

cc @pivotal-nader-ziada  @bobcatfish

### CODE REVIEW

File: docs/developers/README.md [hunk 1]
```diff
@@ -48,9 +46,7 @@ expected in directory path `/workspace/output/resource_name`.
     type: storage
   ```
 
-  - Copy step includes resource being copied to PVC to path
-    `/pvc/task_name/resource_name` from `/workspace/random-space/` if resource
-    is declared in input resource like below example.
+  - If resource is declared both in input and output for task then copy step includes resource being copied to PVC to path `/pvc/task_name/resource_name` from `/workspace/random-space/` if input resource has custom target directory(`random-space`) declared like the following example.
```
Comment (pivotal-nader-ziada) [inline docs/developers/README.md]: In the different examples here, my understanding is that the resource will be copied to the PVC in the path  `/pvc/task_name/resource_name` in all of the cases, if this is correct why do we need 3 examples?

**Comment by shashwathi** [comment]
@pivotal-nader-ziada : https://github.com/knative/build-pipeline/issues/398 for optimization for copy step to PVC

File: docs/developers/README.md (hunk 1 above)
Comment (shashwathi) [reply docs/developers/README.md]: We need 2 examples to show that source path when copied to PVC can change depending on 
1. resource only in output
2. resource in input and output (no target path)
3. resource in input and output( with target path)

**Comment by bobcatfish** [comment]
Thanks for fixing this @shashwathi !

**Pull request merged**